packages/server-renderer/src/helpers/ssrRenderSlot.ts

```diff
@@ -60,7 +62,7 @@ export function ssrRenderSlotInner(
       parentComponent,
       slotScopeId ? ' ' + slotScopeId : '',
     )
-    if (isArray(ret)) {
+    if (isArray(ret) && ensureValidVNode(ret)) {
```

Comment:
This doesn't seem right. Maybe `v-if` should be rendered as `<!---->` like SSR slot. see https://github.com/vuejs/core/blob/5a9626708e970c6fc0b6f786e3c80c22273d126f/packages/compiler-ssr/src/transforms/ssrVIf.ts#L60

Reply:
the proper fix may transform `v-if` to `createElementVNode('',true)` for SSR slot fallback.
https://github.com/vuejs/core/blob/aef807746a178b3ce187bb30b60ff965f01569df/packages/compiler-core/src/transforms/vIf.ts#L230